docs/merkletree/treetex/main.go

```diff
@@ -312,12 +321,47 @@ func modifyRangeNodeInfo() error {
 	return nil
 }
 
+var nodeFormats = map[string]nodeTextFunc{
+	"address": func(id compact.NodeID) string {
+		return fmt.Sprintf("%d.%d", id.Level, id.Index)
+	},
+	"hash": func(id compact.NodeID) string {
+		childLevel := id.Level - 1
+		leftChild := id.Index * 2
+		return fmt.Sprintf("{$H_{%d.%d} = \\\\ H(H_{%d.%d} || H_{%d.%d})$}", id.Level, id.Index, childLevel, leftChild, childLevel, leftChild+1)
+	},
+}
+
 // Whee - here we go!
 func main() {
 	// TODO(al): check flag validity.
 	flag.Parse()
 	height := uint(bits.Len64(*treeSize-1)) + 1
 
+	innerNodeText := nodeFormats[*nodeFormat]
+	if innerNodeText == nil {
+		log.Fatalf("unknown --node_format %s", *nodeFormat)
+	}
+
+	var ldf nodeTextFunc = func(id compact.NodeID) string {
+		if id.Level == 0 {
+			return fmt.Sprintf("%d", id.Index)
```

Comment:
Is it okay that this `Sprintf` does not depend on node format?

Reply:
Yup - that's for the leaves, which defaults to indices unless the user specifies literals with `--leaf_data`